core/src/main/java/org/elasticsearch/index/seqno/GlobalCheckpointSyncAction.java

```diff
@@ -116,13 +117,17 @@ protected void sendReplicaRequest(
     @Override
     protected PrimaryResult<Request, ReplicationResponse> shardOperationOnPrimary(
             final Request request, final IndexShard indexShard) throws Exception {
-        indexShard.getTranslog().sync();
+        if (indexShard.getTranslogDurability() == Translog.Durability.REQUEST) {
+            indexShard.getTranslog().sync();
+        }
         return new PrimaryResult<>(request, new ReplicationResponse());
     }
 
     @Override
     protected ReplicaResult shardOperationOnReplica(final Request request, final IndexShard indexShard) throws Exception {
-        indexShard.getTranslog().sync();
+        if (indexShard.getTranslogDurability() == Translog.Durability.REQUEST) {
```

Comment:
while we're looking at this: Does it make sense to put this into a conditional along the lines of
```
if (indexShard.getTranslog().getLastSyncedGlobalCheckpoint() == globalCheckPoint) {
  indexShard.getTranslog().sync();
}

Reply:
I don't think this buys us anything? We already check this in [`TranslogWriter#syncNeeded`](https://github.com/elastic/elasticsearch/blob/17a2d574de7ad503e7010618c01386f621034c90/core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java#L247)? This is checked [before syncing](https://github.com/elastic/elasticsearch/blob/17a2d574de7ad503e7010618c01386f621034c90/core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java#L327-L329).

Reply:
I discussed this in another channel with @bleskes. The advantage of a guard here is that we avoid fsyncing the translog when the last synced global checkpoint is the current global checkpoint and the translog needs a sync for other reasons.